### PR TITLE
[FEATURE] MultiColumnValues metric

### PR DESCRIPTION
**Pull request by joshua-stauffer**
This PR adds the first multi column values metric, `MultiColumnSumEqualUnexpectedCount`

### CODE REVIEW

File: tests/integration/metrics/multicolumn_values/test_sum_equal.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+COL_A_WITH_NULLS = "A_WITH_NULLS"
+COL_B_WITH_NULLS = "B_WITH_NULLS"
+DATA_FRAME = pd.DataFrame(
+    {
+        COL_A: [1, 2, 3, 4],
+        COL_B: [4, 3, 2, 1],
+    },
+)
+SUCCESS_SUM_TOTAL = 5
+DATA_FRAME_WITH_NULLS = pd.DataFrame(
```
Comment (billdirks) [inline tests/integration/metrics/multicolumn_values/test_sum_equal.py]: This isn't used. Maybe there is a test commit that didn't get pushed up for testing failures? Also it would be nice to test that we catch it when 2 rows don't equal to 5 but neither is `None`.

**Review by billdirks — COMMENTED** [review]
Thanks for doing this. I'd like to see a test for 2 columns not adding up to the target. Also, the None entries aren't being used and it could nice to see if/when they are ignored but maybe that is redundant with the tests in the other metric since they hopefully use the same code path.

File: tests/integration/metrics/multicolumn_values/test_sum_equal.py (hunk 1 above)
Comment (joshua-stauffer) [reply tests/integration/metrics/multicolumn_values/test_sum_equal.py]: yes - i hadn't written the `ignore_row_if` tests yet. While it shares a name with the other `ignore_row_if` condition, it has completely different values, and a separate implementation, so i want to test it.

File: tests/integration/metrics/multicolumn_values/test_sum_equal.py [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        "ignore_row_if,unexpected_count",
+        [
+            pytest.param(
+                "any_value_is_missing", 1, marks=pytest.mark.xfail(reason="returns 6", strict=True)
+            ),
+            pytest.param(
+                "all_values_are_missing",
+                3,
+                marks=pytest.mark.xfail(reason="returns 6", strict=True),
+            ),
```
Comment (NathanFarmer) [inline tests/integration/metrics/multicolumn_values/test_sum_equal.py]: More examples where I think passing `column_types` might help

File: tests/integration/metrics/multicolumn_values/test_sum_equal.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    @pytest.mark.parametrize(
+        "ignore_row_if,unexpected_count",
+        [
+            pytest.param("any_value_is_missing", 1),
+            pytest.param(
+                "all_values_are_missing",
+                3,
+                marks=pytest.mark.xfail(reason="returns 1", strict=True),
+            ),
+            pytest.param("never", 6, marks=pytest.mark.xfail(reason="returns 1", strict=True)),
```
Comment (NathanFarmer) [inline tests/integration/metrics/multicolumn_values/test_sum_equal.py]: Or maybe since this fails on SQL this is a real bug

File: great_expectations/metrics/multi_column_values/multi_column_values.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+from typing import Literal, Optional
```
Comment (NathanFarmer) [inline great_expectations/metrics/multi_column_values/multi_column_values.py]: Can you rename this directory and file `multi_column` to follow the existing pattern

Comment (NathanFarmer) [reply great_expectations/metrics/multi_column_values/multi_column_values.py]: Also the `__init__.py` in this directory should import `from .multi_column import MultiColumnMetric`

File: great_expectations/metrics/multi_column_values/sum.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+from typing import Union
```
Comment (NathanFarmer) [inline great_expectations/metrics/multi_column_values/sum.py]: Can you rename this file `sum_equal.py` like in the tests?

File: tests/integration/metrics/multicolumn_values/test_sum_equal.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
+import pandas as pd
```
Comment (NathanFarmer) [inline tests/integration/metrics/multicolumn_values/test_sum_equal.py]: Can you rename this directory `multi_column`?

**Review by NathanFarmer — CHANGES_REQUESTED** [review]
I'm not blocking on the xfails, just some requests for re-organization. Also, can you add this metric to the `great_expectations.metrics` package?

File: tests/integration/metrics/multicolumn_values/test_sum_equal.py (hunk 3 above)
Comment (joshua-stauffer) [reply tests/integration/metrics/multicolumn_values/test_sum_equal.py]: right, hard to say - i've pushed the explicit types to see if that helps

File: tests/integration/metrics/multi_column/test_sum_equal.py [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+    {
+        COL_A: [1, 2, 3, 4],
+        COL_B: [4, 3, 2, 1],
+    },
+)
+SUCCESS_SUM_TOTAL = 5
+SUCCESS_COUNT = 0
+FAILURE_SUM_TOTAL = 0
+FAILURE_COUNT = 4
+DATA_FRAME_WITH_NULLS = pd.DataFrame(
```
Comment (billdirks) [inline tests/integration/metrics/multi_column/test_sum_equal.py:30]: Can you add a case here where both are not None but don't sum to 5?

**Review by billdirks — APPROVED** [review]
This LGTM, thanks! I have requested you add a test for the failure when the sums do no sum to the expected value and neither are None.

File: tests/integration/metrics/multi_column/test_sum_equal.py (hunk 7 above)
Comment (joshua-stauffer) [reply tests/integration/metrics/multi_column/test_sum_equal.py:30]: i believe the `failure` test case below covers this, but let me know in case there's something else you're looking for

File: tests/integration/metrics/multicolumn_values/test_sum_equal.py (hunk 3 above)
Comment (joshua-stauffer) [reply tests/integration/metrics/multicolumn_values/test_sum_equal.py]: we can't use explicit types as pytest parameters, because they belong to the Spark compatibility layer and need to exist in the top level namespace, and so blow up in the unit tests.

Comment (NathanFarmer) [reply tests/integration/metrics/multicolumn_values/test_sum_equal.py]: I had to do [this](https://github.com/great-expectations/great_expectations/blob/325133d85920cb7bd271d4c65422a8069e560140/tests/integration/data_sources_and_expectations/test_expectation_conditions.py#L26) in the past

**Review by NathanFarmer — APPROVED** [review]

File: great_expectations/metrics/multi_column/multi_column.py [hunk 8]
```diff
@@ -0,0 +1,14 @@
+from typing import Literal, Optional
+
+from great_expectations.compatibility.pydantic import StrictStr
+from great_expectations.expectations.model_field_types import ConditionParser
+from great_expectations.metrics.metric import Metric, _MetricResult
+
+
+class MultiColumnMetric(Metric[_MetricResult], kw_only=True):
```
Comment (NathanFarmer) [inline great_expectations/metrics/multi_column/multi_column.py:8]: I think this should inherit from `BatchMetric`. A `MultiColumnMetric` will always be a subset of a batch. Then you won't need to add `row_condition`. Also, the goal is to infer `condition_parser` eventually, but so far we aren't testing conditions anywhere so I don't care if we include `condition_parser` or not.

**Review by NathanFarmer — CHANGES_REQUESTED** [review]

File: great_expectations/metrics/multi_column/multi_column.py (hunk 8 above)
Comment (joshua-stauffer) [reply great_expectations/metrics/multi_column/multi_column.py:8]: i don't think we can infer `condition_parser`, since for instance the `Spark` allows both the stock GX parser and custom spark queries to be passed through.

**Review by NathanFarmer — APPROVED** [review]
